main/views.py

```diff
@@ -1,57 +1,137 @@
         if match.player1_score > match.player2_score:
             winner = match.player1.name
             loser = match.player2.name
-            w = match.player1_score
-            l = match.player2_score
+            w_score = match.player1_score
+            l_score = match.player2_score
         else:
             winner = match.player2.name
             loser = match.player1.name
-            w = match.player2_score
-            l = match.player1_score
-        results[match.id]={"winner":winner, "loser":loser, "l":l, "w":w}
+            w_score = match.player2_score
+            l_score = match.player1_score
+        results.append({"winner": winner, "loser": loser, "l": l_score, "w": w_score})
     print results
     context = {
         'matches': matches,
         'results': results,
     }
     return render(request, 'main/index.html', context)
 
+
 def players(request):
     players = Player.objects.all()
     context = {
         'players': players,
     }
     return render(request, 'main/players.html', context)
 
+
 def fields(request):
     fields = Field.objects.all()
     context = {
         'fields': fields,
     }
     return render(request, 'main/fields.html', context)
 
+
 def player_detail(request, player_id):
     player = get_object_or_404(Player, pk=player_id)
     return render(request, 'main/player_detail.html', {'player': player})
 
+
 def api_players(request):
-    players = Player.objects.all()
-    data = []
-    for player in players:
-        data.append({"id": player.id, "name": player.name})
-    return JsonResponse({"players": data})
+    if request.method == "GET":
+        players = Player.objects.all()
+        data = []
+        for player in players:
+            data.append({"id": player.id, "name": player.name})
+        return JsonResponse({"players": data})
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        player = Player.objects.create(name=data["name"])
+        return JsonResponse({"id": player.id, "name": player.name})
+
+
+def api_fields(request):
+    if request.method == "GET":
+        fields = Field.objects.all()
+        data = []
+        for field in fields:
+            data.append({"id": field.id, "name": field.name})
+        return JsonResponse({"fields": data})
+    elif request.method == "POST":
+        data = json.loads(request.body)
+        field = Field.objects.create(name=data["name"])
+        return JsonResponse({"id": field.id, "name": field.name})
+
+
+def api_matches(request):
+    if request.method == "GET":
+        matches = Match.objects.all()
+        data = []
+        for match in matches:
+            data.append({"id": match.id, "date_and_time": match.date_and_time,
+                         "player1": match.player1.name, "player2": match.player2.name,
+                         "player1_score": match.player1_score, "player2_score": match.player2_score,
+                         "field": match.field.name})
```

Comment:
Minor: there's a stylistic choice to be made here about the hanging indents. You can either do
```Python
call_function({'foo': 1,
               'bar': 2})
```
as you have here or do
```Python
call_function({
    'foo': 1,
    'bar': 2})
```
or even
```Python
call_function({
    'foo': 1,
    'bar': 2
})
```
All of these are valid [according to PEP8](https://www.python.org/dev/peps/pep-0008/#indentation), so it mostly comes down to personal taste - and consistency. When working alone no one tells you what code style you should implement, but when working in a team, it's helpful to try to stick to a common set of stylistic choices so that it's easy to read each others code. So I'm just letting you know that BIO usually prefers the third type of indentation. But this is not something I want you to 'fix' in your code, just a heads-up.

Reply:
It's also what you are doing in https://github.com/thomasbean-prezi/pongservice/blob/e5fc4f3d56684e9e1e56868a896564eb62ad2b4f/main/views.py#L27-L30